### ISSUE TITLE
UPE - Incorrect payment method shown in admin order details

### ISSUE THREAD

**Issue by diegocurbelo** [woocommerce/woocommerce-gateway-stripe#1933]
**Describe the bug**

After completing a purchase, the customer order page correctly displays the payment method used, but the admin page for the order always says `Credit Card (Stripe)` regardless of the method used:

<img width="559" alt="Screen Shot 2021-09-22 at 11 48 23" src="https://user-images.githubusercontent.com/407542/134395018-3c06fb6d-5145-44d1-8795-a2260a57329e.png">

<img width="559" src="https://user-images.githubusercontent.com/68524302/134352651-d6890204-e5a9-411c-ac28-2b751639f0cc.png">

**To Reproduce**
1. Enable UPE with at least 2 payment methods active.
2. Add a product to the cart
3. Complete the checkout with a non credit card payment method
4. Go to the admin and view the order details as the merchant
5. See that the details say: `Payment via Credit Card (Stripe)` (and not `giropay`)

**Comment by EvanDotPro** [comment]
Oh man, I actually knew this was going to happen but forgot to make an issue for it in the beginning, sorry! Yeah, because we're sharing the one WC payment gateway class for all of the UPE methods, that's what it uses. We'll need to make it use the name of the UPE method class.

(This is actually one reason I originally suggested each UPE method actually being registered as a unique WC payment gateway instead of making method classes "their own thing" like WCPay did, because it's not clear what a clean way to know display either the correct UPE method title or the gateway title option text depending on the context.)

**Comment by bborman22** [comment]
I just did a quick look in to this and just wanted to add some notes of what I found. First, I must have misremembered and WCPay does not update this title either, so perhaps a more generic title here would be sufficient for MVP? Second was the only workaround I could spot was to override the `get_title` function on UPE gateway, but only when on the admin order details page (called from [here](https://github.com/woocommerce/woocommerce/blob/50e8815fa25e8f69e4ca96bbe20d34b0fd321748/includes/admin/meta-boxes/class-wc-meta-box-order-data.php#L198)). Sorry if not adding much, but just wanted to add some details after taking a quick look in case they were helpful at all.
